src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -79,12 +79,17 @@ class AppTasksResource @Inject() (
   @SuppressWarnings(Array("all")) /* async/await */
   def indexTxt(
     @PathParam("appId") appId: String,
+    @DefaultValue("latest")@QueryParam("compatibilityMode") compatibilityMode: String,
     @Context req: HttpServletRequest): Response = authenticated(req) { implicit identity =>
     val id = appId.toRootPath
     result(async {
       val instancesBySpec = await(instanceTracker.instancesBySpec)
       withAuthorization(ViewRunSpec, groupManager.app(id), unknownApp(id)) { app =>
-        ok(EndpointsHelper.appsToEndpointString(instancesBySpec, Seq(app)))
+        val appsToEndpointString = compatibilityMode match {
+          case "1.4" => EndpointsHelper.appsToEndpointStringCompatibleWith14(instancesBySpec, Seq(app))
```

Comment:
magic constants (`"1.4"`) used in multiple places? please use the type system instead